src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Recipes/ElasticSettingsStep.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Recipes.Models;
+using OrchardCore.Recipes.Services;
+using OrchardCore.Search.Elasticsearch.Core.Models;
+using OrchardCore.Search.Elasticsearch.Core.Services;
+
+namespace OrchardCore.Search.Elasticsearch.Core.Recipes
+{
+    /// <summary>
+    /// This recipe step is used to sync Elasticsearch and Lucene settings.
+    /// </summary>
+    public class ElasticSettingsStep : IRecipeStepHandler
+    {
+        private readonly ElasticIndexingService _elasticIndexingService;
+        private readonly ElasticIndexManager _elasticIndexManager;
+
+        public ElasticSettingsStep(
+            ElasticIndexingService elasticIndexingService,
+            ElasticIndexManager elasticIndexManager
+            )
+        {
+            _elasticIndexManager = elasticIndexManager;
+            _elasticIndexingService = elasticIndexingService;
+        }
+
+        public async Task ExecuteAsync(RecipeExecutionContext context)
+        {
+            if (!String.Equals(context.Name, "Settings", StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            var step = context.Step["ElasticSettings"];
```

Comment:
Shouldn't this also be surfaced on the UI as a step?

Reply:
It is the Elasticsearch Search settings. We should probably rename the step title to Elasticsearch settings to remove any confusion.

Reply:
Done,  renamed deployment steps to Lucene Settings and Elasticsearch Settings.